test/test_luaunit.lua

```diff
@@ -1035,6 +1035,33 @@ TestLuaUnitAssertions = { __class__ = 'TestLuaUnitAssertions' }
         assertFailure(lu.assertIsNumber, true)
     end
 
+    function TestLuaUnitAssertions:test_assertIsNaN()
+        assertFailure(lu.assertIsNaN, "hi there!")
+        assertFailure(lu.assertIsNaN, nil)
+        assertFailure(lu.assertIsNaN, {})
+        assertFailure(lu.assertIsNaN, {1,2,3})
+        assertFailure(lu.assertIsNaN, {1})
+        assertFailure(lu.assertIsNaN, coroutine.create( function(v) local y=v+1 end ) )
+        lu.assertIsNaN(0 / 0)
+        assertFailure(lu.assertIsNaN, 0 / 1) -- 0.0
```

Comment:
Let's validate nan with a few more use cases. You can pick from : http://stackoverflow.com/questions/25506281/what-are-all-the-possible-calculations-that-could-cause-a-nan-in-python